web/src/message_edit.js

```diff
@@ -603,9 +605,94 @@ export function start($row, edit_box_open_callback) {
     });
 }
 
+function get_resolve_topic_time_limit_error_string(time_limit, time_limit_unit, topic_is_resolved) {
+    if (topic_is_resolved) {
+        if (time_limit_unit === "minute") {
+            return $t(
+                {
+                    defaultMessage:
+                        "You do not have permission to unresolve topics with messages older than {N, plural, one {# minute} other {# minutes}} in this organization.",
+                },
+                {N: time_limit},
+            );
+        } else if (time_limit_unit === "hour") {
+            return $t(
+                {
+                    defaultMessage:
+                        "You do not have permission to unresolve topics with messages older than {N, plural, one {# hour} other {# hours}} in this organization.",
+                },
+                {N: time_limit},
+            );
+        }
+        return $t(
+            {
+                defaultMessage:
+                    "You do not have permission to unresolve topics with messages older than {N, plural, one {# day} other {# days}} in this organization.",
+            },
+            {N: time_limit},
+        );
+    }
+
+    if (time_limit_unit === "minute") {
+        return $t(
+            {
+                defaultMessage:
+                    "You do not have permission to resolve topics with messages older than {N, plural, one {# minute} other {# minutes}} in this organization.",
+            },
+            {N: time_limit},
+        );
+    } else if (time_limit_unit === "hour") {
+        return $t(
+            {
+                defaultMessage:
+                    "You do not have permission to resolve topics with messages older than {N, plural, one {# hour} other {# hours}} in this organization.",
+            },
+            {N: time_limit},
+        );
+    }
+    return $t(
+        {
+            defaultMessage:
+                "You do not have permission to resolve topics with messages older than {N, plural, one {# day} other {# days}} in this organization.",
+        },
+        {N: time_limit},
+    );
+}
```

Comment:
This looks correct but indeed a bit verbose. @alya I think the main optimization we could do is deciding we don't need separate strings for "unresolve topics".

Reply:
Checked with Alya and she thinks it's worth it, so integrating.